### PR TITLE
Mock Harbor requests

### PR DESCRIPTION
**Pull request by javiermanzo**

### CODE REVIEW

File: Sources/Harbor/Mock/HMocker.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+//  HMocker.swift
+//  Harbor
+//
+//  Created by Javier Manzo on 06/11/2024.
+//
+
+import Foundation
+
+@HRequestManagerActor
+internal final class HMocker: Sendable {
```
Comment (OmarJalil) [inline Sources/Harbor/Mock/HMocker.swift:11]: There is no need to write internal, they are internal by default

File: Sources/Harbor/Mock/HMock.swift [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+//
+
+import Foundation
+
+@HRequestManagerActor
+public struct HMock: Sendable {
+    var request: HRequestBaseRequestProtocol.Type
+    var statusCode: Int
+    var jsonResponse: String?
+    var error: HRequestError?
```
Comment (OmarJalil) [inline Sources/Harbor/Mock/HMock.swift:15]: do these variables need to be vars? can we use them as let?

File: Sources/Harbor/Mock/HMocker.swift [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+//  HMocker.swift
+//  Harbor
+//
+//  Created by Javier Manzo on 06/11/2024.
+//
+
+import Foundation
+
+@HRequestManagerActor
+final class HMocker: Sendable {
```
Comment (OmarJalil) [inline Sources/Harbor/Mock/HMocker.swift:11]: We might want to create an actor for this, maybe we can try to add and remove all mocks at the same time, and boom, race condition

File: Sources/Harbor/Mock/HMock.swift [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+//  HMock.swift
+//  Harbor
+//
+//  Created by Javier Manzo on 06/11/2024.
+//
+
+import Foundation
+
+@HRequestManagerActor
+public struct HMock: Sendable {
```
Comment (OmarJalil) [inline Sources/Harbor/Mock/HMock.swift:11]: There is no need to add sendable in structs if all properties are not modifiable

File: Sources/Harbor/Mock/HMock.swift [hunk 5]
```diff
@@ -0,0 +1,29 @@
+//
+//  HMock.swift
+//  Harbor
+//
+//  Created by Javier Manzo on 06/11/2024.
+//
+
+import Foundation
+
+@HRequestManagerActor
```
Comment (OmarJalil) [inline Sources/Harbor/Mock/HMock.swift:10]: Was this necessary?

Comment (javiermanzo) [reply Sources/Harbor/Mock/HMock.swift:10]: If I dont use the actor, I get this error 
Screenshot 2024-11-13 at 3 21 28 PM

**Review by OmarJalil — APPROVED** [review]

**Pull request merged**